tests/Behat/Gherkin/ParserTest.php

```diff
@@ -171,15 +155,22 @@ protected function parseEtalon($etalon)
         );
     }
 
-    public function testParsingManyCommentsShouldPass()
+    public function testParsingManyCommentsShouldPass(): void
     {
         if (!extension_loaded('xdebug')) {
             $this->markTestSkipped('xdebug extension must be enabled.');
         }
-        $defaultPHPSetting = 256;
-        $this->iniSet('xdebug.max_nesting_level', $defaultPHPSetting);
 
-        $lineCount = 150; // 119 is the real threshold, higher just in case
-        $this->assertNull($this->getGherkinParser()->parse(str_repeat("# \n", $lineCount)));
+        $oldMaxNestingLevel = ini_set('xdebug.max_nesting_level', 256);
+        if ($oldMaxNestingLevel === false) {
+            throw new \RuntimeException('Could not set INI setting value');
+        }
+
+        try {
+            $lineCount = 150; // 119 is the real threshold, higher just in case
+            $this->assertNull($this->getGherkinParser()->parse(str_repeat("# \n", $lineCount)));
+        } finally {
+            ini_set('xdebug.max_nesting_level', $oldMaxNestingLevel);
+        }
```

Comment:
if this is the only place we're customising ini values, I prefer the inline syntax you have here, I think it's clearer than using the helper function in your first commit.